examples/grpc-server-with-prometheus/server/server.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	"github.com/grpc-ecosystem/go-grpc-prometheus"
+	pb "github.com/grpc-ecosystem/go-grpc-prometheus/examples/grpc-server-with-prometheus/protobuf"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// DemoServiceServer defines a Server.
+type DemoServiceServer struct{}
+
+func newDemoServer() *DemoServiceServer {
+	return &DemoServiceServer{}
+}
+
+// SayHello implements a interface defined by protobuf.
+func (s *DemoServiceServer) SayHello(ctx context.Context, request *pb.HelloRequest) (*pb.HelloResponse, error) {
+	pushCounter.Inc()
+	return &pb.HelloResponse{Message: fmt.Sprintf("Hello %s", request.Name)}, nil
+}
+
+// Create a counter metric
+var (
+	pushCounter = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "repository_pushes",
+		Help: "Number of pushes to external repository.",
+	})
+)
+
+func init() {
+	// Metrics have to be registered to be exposed:
+	prometheus.MustRegister(pushCounter)
+}
+
+func main() {
+	// Listen an actual port.
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", 9093))
+	if err != nil {
+		log.Fatalf("failed to listen: %v", err)
+	}
+
+	// Create a HTTP server for prometheus.
+	httpServer := &http.Server{Handler: promhttp.Handler(), Addr: fmt.Sprintf("0.0.0.0:%d", 9092)}
+
+	// Create a gRPC Server with gRPC interceptor.
+	grpcServer := grpc.NewServer(
+		grpc.StreamInterceptor(grpc_prometheus.StreamServerInterceptor),
```

Comment:
Can we be a good example and use a non-global metrics registry? :slightly_smiling_face: As described here: https://github.com/grpc-ecosystem/go-grpc-prometheus/issues/37#issuecomment-374179339

Reply:
Good idea, but it will be hard to convince people to use that pattern (:  

But we can try...

Reply:
Sure. This pattern looks like simple and awesome. But i propose that don't add this pattern in  the example. Because this parttern is difficult to understand to beginner.

Reply:
Yeah I agree, but we should still do our part, similar to how we should teach the world not to use global loggers. It’s sad that we introduced the global pattern in the first place, but let’s make sure we encourage the best use whenever we can! :)

Reply:
yea.. The truth is @fengzixu that we should not use global at all ): Even beginners should not ^^

Reply:
Is the example given above correct ? (https://github.com/grpc-ecosystem/go-grpc-prometheus/issues/37#issuecomment-374179339) I try to add it to my example, but not running.

Reply:
@brancz

Reply:
Yeah I will fix it according https://github.com/grpc-ecosystem/go-grpc-prometheus/issues/37#issuecomment-374179339